packages/iris-grid/src/mousehandlers/IrisGridContextMenuHandler.tsx

```diff
@@ -717,44 +717,56 @@ class IrisGridContextMenuHandler extends GridMouseHandler {
 
     const actions: ResolvableContextAction[] = [];
 
-    if (modelColumn != null && modelRow != null) {
-      const value = model.valueForCell(modelColumn, modelRow);
-
-      const valueText = model.textForCell(modelColumn, modelRow);
-      const column = columns[modelColumn];
+    if (modelColumn != null) {
+      const sourceCell = model.sourceForCell(modelColumn, modelRow ?? 0);
+      const { column: sourceColumn, row: sourceRow } = sourceCell;
+      if (modelRow != null) {
+        const value = model.valueForCell(sourceColumn, sourceRow);
+        const valueText = model.textForCell(sourceColumn, sourceRow);
+        const column = columns[sourceColumn];
 
-      const { onContextMenu } = irisGrid.props;
+        const { onContextMenu } = irisGrid.props;
 
-      if (column != null) {
-        actions.push(
-          ...onContextMenu({
-            model,
-            value,
-            valueText,
-            column,
-            rowIndex,
-            columnIndex,
-            modelRow,
-            modelColumn,
-          })
-        );
+        if (column != null) {
+          actions.push(
+            ...onContextMenu({
+              model,
+              value,
+              valueText,
+              column,
+              rowIndex,
+              columnIndex,
+              modelRow,
+              modelColumn,
+            })
+          );
+        }
       }
-    }
 
-    if (modelColumn != null && model.isFilterable(modelColumn)) {
-      // Clear column filter should still be available after last row
-      // And should be available in both header and body context menus
-      actions.push({
-        title: 'Clear Column Filter',
-        group: IrisGridContextMenuHandler.GROUP_FILTER,
-        order: 30,
-        action: () => {
-          this.irisGrid.removeColumnFilter(modelColumn);
-        },
-        disabled: !(
-          quickFilters.has(modelColumn) || advancedFilters.has(modelColumn)
-        ),
-      });
+      const clearFilterRange = model.getClearFilterRange(modelColumn);
+      if (clearFilterRange != null && clearFilterRange.length > 0) {
+        // Clear column filter should still be available after last row
+        // And should be available in both header and body context menus
+        actions.push({
+          title:
+            clearFilterRange[1] - clearFilterRange[0] > 0
+              ? 'Clear Group Filter'
+              : 'Clear Column Filter',
+          group: IrisGridContextMenuHandler.GROUP_FILTER,
+          order: 30,
+          action: () => {
+            this.irisGrid.removeColumnFilter(clearFilterRange);
+          },
+          disabled: !(
+            Array.from(quickFilters.keys()).some(
+              col => col >= clearFilterRange[0] && col <= clearFilterRange[1]
+            ) ||
+            Array.from(advancedFilters.keys()).some(
+              col => col >= clearFilterRange[0] && col <= clearFilterRange[1]
+            )
+          ),
```

Comment:
Minor, but logic here could be flipped to `!A && !B` instead of `!(A || B)`. Just a little easier to understand